### PR TITLE
Ignore removal of non-Calico chain.

### LINKED ISSUE #1349: Under churn with k8s, golang DP driver thinks KUBE-SEP chain out-of-sync
Under churn tests, we're seeing logs like this:
```
2017-02-16 22:24:01.608 [WARNING][32] table.go 444: Detected out-of-sync inserts, marking for resync actualRuleIDs=[]string(nil) chainName="KUBE-SEP-WE2O5CDAE2MDKF6F" expectedRuleIDs=[]string{} ipVersion=0x4 table="nat"
```
_Probably_ benign since the later stage of processing will spot that there's nothing to do but we should squash this false positive.

### PR DESCRIPTION
**Pull request by fasaxc**
Fixes #1349

### CODE REVIEW

File: iptables/table_test.go [hunk 1]
```diff
@@ -921,3 +921,51 @@ func describeDirtyDataplaneTests(appendMode bool) {
...
+			dataplane.Chains = map[string][]string{
+				"FORWARD": {"-m comment --comment \"cali:hecdSCslEjdBPBPo\" --jump DROP"},
+			}
+			table.Apply()
+		})
+
+		It("should ignore the deletion", func() {
+			Expect(dataplane.Chains).To(Equal(map[string][]string{
+				"FORWARD": {"-m comment --comment \"cali:hecdSCslEjdBPBPo\" --jump DROP"},
+			}))
```
Comment (neiljerram) [inline iptables/table_test.go]: Shouldn't we also test here that there isn't any refresh processing?

Comment (fasaxc) [inline iptables/table_test.go]: I couldn't see a way to verify that without adding a test-only flag of some sort.  Any ideas?

Comment (neiljerram) [inline iptables/table_test.go]: I thought that was what the infrastructure like

			dataplane.ResetCmds()

and

		Expect(dataplane.CmdNames).To(BeEmpty())

was for.  No?

Comment (fasaxc) [inline iptables/table_test.go]: Ah yep, I was thinking of how to make sure the old warning didn't get hit.

**Pull request merged**